pkg/token/joinclient_test.go

```diff
@@ -142,7 +143,7 @@ func TestJoinClient_Cancellation(t *testing.T) {
 }
 
 func startFakeJoinServer(t *testing.T, handler func(http.ResponseWriter, *http.Request)) (*url.URL, []byte) {
-	requestCtx, cancelRequests := context.WithCancel(context.Background())
+	requestCtx, cancelRequests := context.WithCancel(t.Context())
```

Comment:
```suggestion
```

And replace `BaseContext` on line 178 with `func(net.Listener) context.Context { return t.Context() },`